spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubOperations.java

```diff
@@ -16,14 +16,39 @@
 
 package org.springframework.cloud.gcp.pubsub.core;
 
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.cloud.pubsub.v1.Subscriber;
+
 import org.springframework.messaging.Message;
 import org.springframework.util.concurrent.ListenableFuture;
 
 /**
+ * An abstraction for Google Cloud Pub/Sub.
+ *
  * @author Vinicius Carvalho
+ * @author João André Martins
  */
 public interface PubSubOperations {
 
-	ListenableFuture<String> send(String topic, Message message);
+	/**
+	 * Sends a Spring {@link Message} to Pub/Sub.
+	 *
+	 * @param topic the name of an existing topic
+	 * @param message the Spring message
+	 * @return the listenable future of the call
+	 */
+	ListenableFuture<String> publish(String topic, Message message);
+
+	/**
+	 * Polls a subscription for new messages.
+	 *
+	 * <p>
+	 * The created {@link Subscriber} is returned so it can be stopped.
+	 *
+	 * @param subscription the name of an existing subscription
+	 * @param messageHandler the callback method activated whenever a new message arrives
+	 * @return subscriber listening to new messages
+	 */
+	Subscriber poll(String subscription, MessageReceiver messageHandler);
```

Comment:
I'd recommend calling this `subscribe`.  Poll indicates that we are pulling/polling for new messages. But that may not be true.

Reply:
I had `subscribe` at the beginning, but then changed.
`subscribe` isn't the right term, because you subscribe to a topic and the argument here is `subscription`.
How would polling not happen here? We start and return a subscriber, it's bound to poll. New messages may or may not arrive, but polling is happening regardless.

Reply:
Well, `poll` in most cases mean one time operation. See Apache Kafka and Spring Integration Polling Channel Adapter. 
`subscribe` is definitely good to determine that we are about long-life repeatable process. I can suggest a `consume` term. 
However the `MessageReceiver` argument may definitely be our saver and together with a good JavaDoc we are good to go with out `poll` variant.
WDYT?

Reply:
From the perspective of the `MessageHandler`, it's more like a subscribe (polling is an implementation detail). It's arguably the parameter that this method is most focused on since the subscription String is just a name. So I agree with @saturnism FWIW.

Reply:
If the subscriber is a push subscriber, then you're right that polling doesn't happen, so this should definitely be renamed. (It's the only time polling doesn't happen, as our Pub/Sub Java API only supports polling/async pulling AFAIK)

I still have some reservations about `subscribe`, though. Conceptually, you subscribe to a topic, whereas here we're subscribing to a subscription which doesn't make much sense?

`subscribe` would be fine if the argument was `topic` instead of `subscription` and `PubSubTemplate` created a subscription for that topic. However, this is a little out of the scope of this PR. I'll reticently go for `subscribe` due to public demand.